engine/src/main/java/io/zeebe/engine/processor/workflow/job/JobBatchActivateProcessor.java

```diff
@@ -221,4 +236,31 @@ private void rejectCommand(
     streamWriter.appendRejection(record, rejectionType, rejectionReason);
     responseWriter.writeRejectionOnCommand(record, rejectionType, rejectionReason);
   }
+
+  private void raiseIncidentJobTooLargeForMessageSize(
+      long jobKey, final JobRecord job, final TypedStreamWriter streamWriter) {
+
+    final DirectBuffer jobErrorMessage = job.getErrorMessageBuffer();
+    DirectBuffer incidentErrorMessage =
+        wrapString(
+            String.format(
+                "The job with key '%s' is too large for the configured message size.", jobKey));
+    if (jobErrorMessage.capacity() > 0) {
+      incidentErrorMessage = jobErrorMessage;
+    }
```

Comment:
The job should not have an error message before. So, we can directly set the incident message. 

I think it could help the user if we provide more information about the message size and how to fix it. For example, 

`The job with key '123' can not be activated because it is larger than the configured message size (1,8MB). Try to reduce the size by limiting the fetched variables or modifying the variable values.`.